src/Fmt_ast.ml

```diff
@@ -1970,26 +1970,48 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                $ fmt_expression c (sub_exp ~ctx e2) )
            $ fmt_atrs ))
   | Pexp_sequence _ ->
+      let blank_line xexp1 xexp2 =
+        match c.conf.sequence_blank_line with
+        | `Preserve_one ->
+            let l1 = xexp1.ast.pexp_loc.loc_end.pos_lnum in
+            let l2 = xexp2.ast.pexp_loc.loc_start.pos_lnum in
+            l2 - l1 > 1
+        | `Compact -> false
+      in
       let is_simple x = is_simple c.conf width x in
       let break xexp1 xexp2 = not (is_simple xexp1 && is_simple xexp2) in
       let grps = List.group (Sugar.sequence c.cmts xexp) ~break in
       let fmt_grp grp =
-        list grp
-          ( match c.conf.sequence_style with
-          | `Separator when c.conf.break_sequences -> " ;@;<1000 0>"
-          | `Separator -> " ;@ "
-          | `Terminator when c.conf.break_sequences -> ";@;<1000 0>"
-          | `Terminator -> ";@ " )
-          (fmt_expression c)
+        list_pn grp (fun ?prev x ?next:_ ->
+            ( match prev with
+            | Some prev -> (
+              match c.conf.sequence_style with
```

Comment:
Minor comment: The matched value only change the leading space so there is code duplication. The break logic should be factorized. Eg.
```ocaml
let break =
  if blank_line prev x then fmt "\n@;<1000 0>"
  else if c.conf.break_sequences then fmt "@;<1000 0>"
  else fmt "@ "
in
match c.conf.sequence_style with
| `Separator -> str " ;" $ break
| `Terminator -> str ";" $ break
```

Also, there is a very similar function a few lines below, could they share the same code ? (eg. with some sort of `force_break_sequence` bool argument)

Reply:
done